code/modules/events/pirate.dm

```diff
@@ -0,0 +1,91 @@
+#define LOC_CARGO		0
+#define LOC_SECURITY	1
+#define LOC_RESEARCH	2
+#define LOC_HALLWAYS	3
+#define LOC_BRIDGE		4
+
+/datum/event/pirate
+	var/location
+	var/locstring
+	var/spawn_area_type
+	var/spawncount = 1
+	var/boss_spawn_count = 1
+	var/cloud_hueshift
+	var/list/players = list()
+	has_skybox_image = FALSE
+
+/datum/event/pirate/setup()
+	announceWhen = rand(announceWhen, announceWhen + 3)
+	startWhen = announceWhen
+	endWhen = 45
+
+/datum/event/pirate/announce()
+	command_announcement.Announce("Attention Landlubers of the [station_name()], hand over your booty and your ship and no one gets hurt!", "Incoming Transmission:", new_sound = sound('sound/effects/siren.ogg', volume=25))
+
+/datum/event/pirate/start()
+	location = rand(0,4)
+	switch(location)
+		if(LOC_CARGO)
+			spawn_area_type = /area/quartermaster
+			locstring = "cargo & mining"
+			spawncount = rand(1 * severity, 3 * severity)
+			boss_spawn_count = 1
+		if(LOC_SECURITY)
+			spawn_area_type = /area/security
+			locstring = "security"
+			spawncount = rand(1 * severity, 3 * severity)
+			boss_spawn_count = rand(1,2)
+		if(LOC_RESEARCH)
+			spawn_area_type = /area/rnd
+			locstring = "research and development"
+			spawncount = rand(1 * severity, 3 * severity)
+			boss_spawn_count = rand(1,2)
+		if(LOC_HALLWAYS)
+			spawn_area_type =  /area/hallway
+			locstring = "public hallways"
+			spawncount = rand(2 * severity, 4 * severity)
+			boss_spawn_count = rand(1,2)
+		if(LOC_BRIDGE)
+			spawn_area_type = /area/bridge
+			locstring = "bridge"
+			spawncount = rand(1 * severity, 3 * severity)
+			boss_spawn_count = rand(1,2)
+
+/datum/event/pirate/end()
+	var/list/vents = list()
+	for(var/areapath in typesof(spawn_area_type))
+		var/area/A = locate(areapath)

```

Comment:
whhhhhhhhhhhy is this using vents?
also locate(areapath) won't work
you're going to need to either use a helper or use GLOB.sortedAreas (GLOB.areas_by_type exists but you don't know if it's unique or not so probably don't use that)

Reply:
It was based on how existing enemy spawning events already work. I am not sure what else to really use, unless there is some way to check if all the tiles in the room are unobstructed by machines, structures, windows so the mobs don't spawn in them. Vents work well for this as at worst they are place in unobstructed spots or at worse under tables. How would you suggest I set up a helper and or spawn the mobs otherwise?

Reply:
> whhhhhhhhhhhy is this using vents?
> also locate(areapath) won't work
> you're going to need to either use a helper or use GLOB.sortedAreas (GLOB.areas_by_type exists but you don't know if it's unique or not so probably don't use that)
It was based on how existing enemy spawning events already work. I am not sure what else to really use, unless there is some way to check if all the tiles in the room are unobstructed by machines, structures, windows so the mobs don't spawn in them. Vents work well for this as they are placed in unobstructed spots or at worse under tables. How would you suggest I set up a helper and or spawn the mobs otherwise?

Reply:
Oh second thing this also means the pirates even with their limit spawns tend to group up in areas around vents rather than being spread almost evenly around the area.